model/lifecycle.go

```diff
@@ -373,47 +396,42 @@ func RestartVersion(versionId string, taskIds []string, abortInProgress bool, ca
 
 	// Only need to check one task per task group / build combination
 	taskGroupsToCheck := map[taskGroupAndBuild]task.Task{}
-	tasksToRestart := allFinishedTasks
-
 	restartIds := []string{}
-	for _, t := range tasksToRestart {
+	for _, t := range allFinishedTasks {
 		if t.IsPartOfSingleHostTaskGroup() {
-			if err = t.SetResetWhenFinished(); err != nil {
+			if err := t.SetResetWhenFinished(); err != nil {
 				return errors.Wrapf(err, "marking '%s' for restart when finished", t.Id)
 			}
 			taskGroupsToCheck[taskGroupAndBuild{
 				Build:     t.BuildId,
 				TaskGroup: t.TaskGroup,
 			}] = t
-		} else {
-			// Only hard restart non-single host task group tasks
-			restartIds = append(restartIds, t.Id)
-			if t.DisplayOnly {
-				restartIds = append(restartIds, t.ExecutionTasks...)
-			}
+			continue
```

Comment:
preference to keep this an if / else statement; continue seems more useful when we're exiting early in a function that does a lot of stuff, or when the only thing we want to do in the if case is continue